zerver/lib/markdown/api_return_values_table_generator.py

```diff
@@ -61,6 +76,15 @@ def render_table(self, return_values: Dict[str, Any], spacing: int) -> List[str]
         for return_value in return_values:
             if return_value in IGNORE:
                 continue
+            if 'oneOf' in return_values[return_value]:
+                ans.append(self.render_desc(return_values[return_value]['description'],
+                                            spacing, return_value))
+                for element in return_values[return_value]['oneOf']:
+                    if 'description' not in element:
+                        continue
+                    ans.append(self.render_desc(element['description'], spacing + 4))
+                    ans += self.render_table(element['properties'], spacing + 8)
+                continue
```

Comment:
I don't understand what the block does -- can you comment/explain it?

Reply:
Added comments.
```
            if 'oneOf' in return_values[return_value]:
                # For elements using oneOf there are two descriptions. The first description
                # should be at level with the oneOf and should containg the basic non-specific
                # description of the endpoint. Then for each element of oneOf there is a
                # specialized description for that particular case. The description used
                # right below is the main description.
                ans.append(self.render_desc(return_values[return_value]['description'],
                                            spacing, return_value))
                for element in return_values[return_value]['oneOf']:
                    if 'description' not in element:
                        continue
                    # Add the specialized description of the oneOf element.
                    ans.append(self.render_desc(element['description'], spacing + 4))
                    # If the oneOf element is an object schema then render the documentation
                    # of its keys.
                    if 'properties' in element:
                        ans += self.render_table(element['properties'], spacing + 8)
                continue
```python